src/os/filestore/FileStore.cc

```diff
@@ -3754,18 +3754,22 @@ int FileStore::_clone_range(const coll_t& cid, const ghobject_t& oldoid, const g
 }
 
 /*
- * Move contents of temp object according to move_info to base object. Once the move_info is traversed completely, delete the temp object.
+ * Move contents of src object according to move_info to base object. Once the move_info is traversed completely, delete the src object.
  */
-int FileStore::_move_tempobjs(const coll_t& src_cid, const ghobject_t& src_oid, const coll_t& cid, const ghobject_t& oid,
+int FileStore::_merge_delete_srcobj(const coll_t& src_cid, const ghobject_t& src_oid, const coll_t& cid, const ghobject_t& oid,
                               const vector<boost::tuple<uint64_t, uint64_t, uint64_t>> move_info,
                               const SequencerPosition& spos)
 {
-//after all processing is done, delete the tempobj.
   int r = 0;
   bool err = false;
+  int dstcmp;
   FDRef t, b;
 
-  dout(10) << "move_tempobjs " << src_cid << "/" << src_oid << " -> " << cid << "/" << oid << dendl;
+  dout(10) << "merge_delete_srcobj " << src_cid << "/" << src_oid << " -> " << cid << "/" << oid << dendl;
+
+  dstcmp = _check_replay_guard(cid, oid, spos);
+  if (dstcmp < 0)
```

Comment:
Is it possible for dstcmp here to be 0?  Can you be sure the source was removed if dstcmp is -1?